Steps/UploadCodeCoverage.yml

```diff
@@ -0,0 +1,83 @@
+## @file
+# Azure Pipelines step template upload code coverage to codecov.io
+#
+# Follows the codecov.io documentation for uploading code coverage reports:
+# https://docs.codecov.com/docs/codecov-uploader
+#
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# SPDX-License-Identifier: BSD-2-Clause-Patent
+##
+
+parameters:
+- name: codecov_token
+  displayName: Codecov.io Token
+  type: string
+  default: ''
+- name: report_dir
+  displayName: Code Coverage Report
+  type: string
+  default: ''
+
+steps:
+- task: PythonScript@0
+  displayName: Download and Verify Codecov Uploader
+  inputs:
+    scriptSource: inline
+    script: |
+      import platform
+      import requests
```

Comment:
what is the pip module that installs requests? do we need to add it to pip-requirements?

Reply:
I've updated it such that it will automatically install the latest version of requests. It can also be turned off by using the parameter `install_dependencies: false`